client/src/app/core/repositories/topics/topic-repository.service.ts

```diff
@@ -91,14 +85,35 @@ export class TopicRepositoryService extends BaseIsAgendaItemAndListOfSpeakersCon
         return this.translate.instant(plural ? 'Topics' : 'Topic');
     };
 
-    public duplicateTopic(topic: ViewTopic): void {
-        this.create(
-            new CreateTopic({
-                ...topic.topic,
-                agenda_type: topic.agenda_item.type,
-                agenda_parent_id: topic.agenda_item.parent_id,
-                agenda_weight: topic.agenda_item.weight
-            })
+    public duplicateTopic(topicAgendaItem: ViewAgendaItem): Promise<Identifiable> {
+        return this.create(this.getDuplicatedTopic(topicAgendaItem));
+    }
+
+    public duplicateMultipleTopics(agendaItems: ViewAgendaItem[]): Promise<void> {
+        const payload: TopicAction.CreatePayload[] = agendaItems.map(item =>
+            this.getCreatePayload(this.getDuplicatedTopic(item))
         );
+        return this.sendBulkActionToBackend(TopicAction.CREATE, payload);
+    }
+
+    private getDuplicatedTopic(topicAgendaItem: ViewAgendaItem): CreateTopic {
+        const viewTopic = topicAgendaItem.content_object as ViewTopic;
+        return new CreateTopic({
```

Comment:
Still not a fan of create models. Please use the createAction interface for this. Maybe you are able to completely remove `CreateTopic` (it may be used during importing...).

Reply:
Yes, I'm not, too. There are many changes, that are not pushed yet. Those "*-create-*"-files will be completely deleted. :)

Reply:
Yeah!